editor/src/templates/editor.tsx

```diff
@@ -127,15 +129,43 @@ function replaceLoadingMessage(newMessage: string) {
   }
 }
 
+function collectElementsToRerenderTransient(
+  working: Array<ElementPath>,
+  action: EditorAction,
+): Array<ElementPath> {
+  if (action.action === 'TRANSIENT_ACTIONS') {
+    if (action.elementsToRerender != null) {
+      working.push(...action.elementsToRerender)
+    }
+    working.push(...action.transientActions.reduce(collectElementsToRerenderTransient, working))
+    return working
+  } else {
+    return working
+  }
+}
+
 // If the elements to re-render have specific paths in 2 consecutive passes, but those paths differ, then
 // for this pass treat it as `rerender-all-elements`, to ensure that the metadata gets cleaned up as
 // the previously focused elements may not now exist.
 // Also as some canvas strategies may not supply a specific set of elements to re-render, if
 // `rerender-all-elements` switches to a specific set of paths, ignore the specific set of paths
 // for the very first pass to get another `rerender-all-elements`.
 let lastElementsToRerender: ElementsToRerender = 'rerender-all-elements'
-function fixElementsToRerender(currentElementsToRerender: ElementsToRerender): ElementsToRerender {
-  let elementsToRerender: ElementsToRerender = currentElementsToRerender
+function fixElementsToRerender(
+  currentElementsToRerender: ElementsToRerender,
+  dispatchedActions: readonly EditorAction[],
+): ElementsToRerender {
+  // while running transient actions there is an optional elementsToRerender
+  const elementsToRerenderTransient = uniqBy<ElementPath>(
```

Comment:
I think I'd clean the name here too
```suggestion
  const elementsToRerenderFromTransientActions = uniqBy<ElementPath>(
```